### PR TITLE
Pass leaderboard and statistics as modals

### PR DESCRIPTION
**Pull request by JEANPRBT**
This PR passes both leaderboard and statistics as modals, accessible everywhere in the app. 
It also adds small animation on infobox hint reveal.

### CODE REVIEW

File: presenters/GamePresenter.vue [hunk 1]
```diff
@@ -45,29 +50,37 @@ function guessAndCheck(name : string){
...
+      const aStats = a.stats.find((stat: any) => parseInt(stat.date) === getCurrentDayTimestamp());
+      const bStats = b.stats.find((stat: any) => parseInt(stat.date) === getCurrentDayTimestamp());
+      if (aStats && bStats) {
+        if (aStats.guesses === bStats.guesses) {
+          return aStats.time - bStats.time;
+        }
+        return aStats.guesses - bStats.guesses;
       }
-      return aStats.nbGuesses - bStats.nbGuesses;
+      return 0;
```
Comment (roxannecvl) [inline presenters/GamePresenter.vue:71]: if (aStats && bStats) should be removed as it is already in the filter

File: presenters/GamePresenter.vue [hunk 2]
```diff
@@ -76,7 +89,7 @@ async function computeRank(){
 }
 
 function hasPlayedAtDate(item : any, timestamp: number){
-  return item.timedStats !== undefined && item.timedStats.find((stat: any) => parseInt(stat.date) === timestamp)
+  return
 }
```
Comment (roxannecvl) [inline presenters/GamePresenter.vue:93]: remove

File: views/StatisticsView.vue [hunk 3]
```diff
@@ -35,6 +37,9 @@ const props = defineProps({
 // Emits
 const emit = defineEmits(['populate-stats'])
```
Comment (roxannecvl) [inline views/StatisticsView.vue:39]: keep this but remove the function populateStats and emit directly

**Review by roxannecvl — APPROVED** [review]
Fix the three comments before merging

**Review by martinctl — APPROVED** [review]

**Pull request merged**